static/app/views/alerts/rules/metric/details/metricHistory.tsx

```diff
@@ -43,92 +32,85 @@ function MetricAlertActivity({organization, incident}: MetricAlertActivityProps)
   const activities = (incident.activities ?? []).filter(
     activity => activity.type === IncidentActivityType.STATUS_CHANGE
   );
-  const criticalActivity = activities.filter(
+  const criticalActivity = activities.find(
     activity => activity.value === `${IncidentStatus.CRITICAL}`
   );
-  const warningActivity = activities.filter(
+  const warningActivity = activities.find(
     activity => activity.value === `${IncidentStatus.WARNING}`
   );
 
-  const triggeredActivity = criticalActivity.length
-    ? criticalActivity[0]
-    : warningActivity[0];
-  const currentTrigger = getTriggerName(triggeredActivity.value);
+  const triggeredActivity: ActivityType = criticalActivity
+    ? criticalActivity!
+    : warningActivity!;
```

Comment:
Is it ever possible that there is no triggered activity?

Reply:
I don't think so. it was using `.filter` with `criticalActivity[0]` before which our typescript just isn't setup to check if its undefined